python/lsst/daf/butler/registry/interfaces/database.py

```diff
@@ -330,9 +326,11 @@ def transaction(self, *, savepoint: bool = False, interrupting: bool = False) ->
             "Logic error in transaction nesting: an operation that would "
             "interrupt the active transaction context has been requested."
         )
-        if savepoint:
+        if self._cs.connection.in_transaction():
```

Comment:
I guess this means we treat all nested transactions as savepoints. Savepoints have bigger overhead compared to non-savepoint "nested" sqlalchemy transactions, if we have a lot of nested transaction in registry this may cause performance issues.

Reply:
Yup, understood.  As this is already a problem in the current `SqlRegistry`, I'm just punting the problem to a later ticket (I have some other mitigations in mind) as it's too tricky to solve quickly on this one.